src/bin/cli.rs

```diff
@@ -16,36 +16,45 @@
 
 extern crate clap;
 extern crate csv;
-extern crate env_logger;
 extern crate failure;
 extern crate good_stv;
-#[macro_use]
-extern crate log;
 extern crate rand;
+#[macro_use]
+extern crate slog;
+extern crate slog_async;
+extern crate slog_term;
 
 use std::io;
 
-use clap::{App, Arg};
+use clap::{App, Arg, ArgMatches};
 use failure::{Error, ResultExt};
+use slog::Drain;
 
 use good_stv::*;
 
 const VERSION: Option<&str> = option_env!("CARGO_PKG_VERSION");
 
 fn main() {
-    if let Err(err) = run() {
-        error!("{:?}", err);
-        for cause in err.causes().skip(1) {
-            error!("Caused by: {:?}", cause);
+    let matches = parse_opts();
+    let exit_code = {
```

Comment:
This slightly awkward block is because with `slog` the logger has to have gone out of scope by the time the program exits, so calling `std::process::exit(1)` while the logger was in scope had to be rewritten.

Reply:
:+1:

Reply:
id possibly wrap the block in the process::exit but that's a style thing